storage/localstore/migration.go

```diff
@@ -161,12 +161,9 @@ func migrateSanctuary(db *DB) error {
 
 	err = db.pushIndex.Iterate(func(item shed.Item) (stop bool, err error) {
 		tag, err := db.tags.Get(item.Tag)
-		if err != nil {
```

Comment:
Would not it be better that if there is not found error, just to continue iteration, and to return an error if it is anything else except nil?

Reply:
i'm not sure i understand. the only error that can happen here is the artificial error we construct in `tags.go`. there's no error coming from the `sync.Map` `Get` method. So my assumption is that if there is no error and the tag is not nil then add to the batch. If we return the tag not found error the migration will fail. This could be the reason for the failures we've been seeing

Reply:
it is also expected that the tags will be empty for nodes upgrading from the previous stable release.

Reply:
This code relies on the current implementation that tag.Get return only one error. This ties this code to the implementation of tags, and any change to tag.Get returning a different error would couple with this code logic. From the point of this package (and any other that uses tags), tags Get may return any error, and that should be handled. If tags.Get would return a boolean whether it is found or not, then we could rely on its state, but error can be anything else then just nil or not nil. A popper error handling is important, as well not relaying on implementation specifics of the imported package, but on the exposed api that it provides.

Reply:
fair enough